### PR TITLE
Update VLC to 3.0.0 & dependencies

### PR DESCRIPTION
**Pull request by MatMaul**
Replace fluidsynth by fluidlight
Replace Samba with libDSM (supports discovery and in official VLC contribs)
Add libplacebo, microdns, spatialaudio, x264, x265, libvpx, srt, mp3lame, libdvdcss
Add fortified build options

### CODE REVIEW

**Comment by TingPing** [comment]
> Replace udev by eudev

Why?

**Comment by MatMaul** [comment]
Because standalone udev hasn't been updated in years since it is now integrated in systemd.
eudev is recent systemd udev extracted from systemd.

File: org.videolan.VLC.json [hunk 1]
```diff
@@ -700,13 +723,13 @@
       ]
     },
     {
-      "name": "libmodplug",
+      "name": "nasm",
```
Comment (TingPing) [inline org.videolan.VLC.json]: `nasm` is in the runtime.

Comment (MatMaul) [reply org.videolan.VLC.json]: The one there is too old for recent x264.

**Comment by TingPing** [comment]
As I understand it `udev` doesn't do anything in the sandbox anyway and the lib just exists to full-fill a build requirement.

**Comment by MatMaul** [comment]
I am not sure. We are exposing the host /dev tree here with --device=all, and udev is perhaps actually used by some of the libs for USB related stuffs, or to read CD/DVDs.
If it really bothers you I am fine with removing the change, I did it while I was having "fun" updating the rest but it is clearly not the most important in this changeset :)

**Comment by TingPing** [comment]
I don't have a strong opinion on it, mostly wondering if we should change all packages to `eudev` if it is a meaningful change.

**Comment by TingPing** [comment]
bot, build

**Comment by MatMaul** [comment]
bot, build

**Comment by MatMaul** [comment]
bot, build

File: org.videolan.VLC.json [hunk 2]
```diff
@@ -39,8 +39,33 @@
...
+    "cxxflags": "-O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2",
+    "ldflags": "-fstack-protector-strong -Wl,-z,relro,-z,now -L/app/lib",
+    "env": {
+      "V": "1"
+    }
+  },
   "modules": [
-    "shared-modules/udev/udev-175.json",
+    {
+      "name": "eudev",
```
Comment (cosimoc) [inline org.videolan.VLC.json]: I'd also keep the original `udev` dependency here. I don't think `eudev` has any advantage inside a sandbox compared to `udev` itself.

Comment (MatMaul) [reply org.videolan.VLC.json]: Fine for me, removed.

**Comment by nedrichards** [comment]
bot, build

**Comment by nedrichards** [comment]
It'd be good to land this before the full 3.0 update - which we want to get in ASAP anyway.

**Comment by MatMaul** [comment]
bot, build

**Comment by MatMaul** [comment]
bot build

**Comment by MatMaul** [comment]
bot, build

**Comment by MatMaul** [comment]
bot, build

**Comment by nedrichards** [comment]
@MatMaul you need to be a repo committer or admin to be able to build, sorry! Although with these sort of PRs you're very much on the way

**Comment by nedrichards** [comment]
bot, build

**Comment by eszlari** [comment]
Am I missing something or shouldn't this be rebased on master? (because it removes SMB support)

**Comment by MatMaul** [comment]
@eszlari what do you mean ? You misread I didn't remove SMB support, I just changed the underlying lib from smbclient to libDSM, which supports discovery and is used in the contribs of VLC.
I can keep smbclient but libDSM will be used anyway, it has a higher priority.

@nedrichards I have no idea about this app id appstream error in the builds, it looks fine to me.

**Comment by MatMaul** [comment]
Nervermind the appstream is indeed wrong now, fix incoming.

**Comment by eszlari** [comment]
>  I can keep smbclient but libDSM will be used anyway, it has a higher priority.

I never heard of libDSM before, that's why I was confused. I think keeping smbclient doesn't make sense.

**Comment by TingPing** [comment]
bot, build

**Pull request merged**